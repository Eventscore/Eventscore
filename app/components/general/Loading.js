//<Loading text={'Server is down'} />
import React, { PropTypes } from 'react';
import { View, ActivityIndicator } from 'react-native';

// Consts and Libs
import { AppStyles } from '@theme/';

// Components
import { Spacer, Text } from '@ui/';

const Loading = ({ text, transparent }) => (
  <View
    style={[
      AppStyles.container,
      AppStyles.containerCentered,
      transparent && { backgroundColor: 'rgba(255,255,255,0.75)' },
    ]}
  >
    <ActivityIndicator
      animating
      size={'large'}
      color={transparent ? '#000' : '#AAA'}
    />

    <Spacer size={10} />

    {!!text && <Text>{text}</Text>}
  </View>
);

Loading.propTypes = { text: PropTypes.string, transparent: PropTypes.bool };
Loading.defaultProps = { text: null, transparent: false };
Loading.componentName = 'Loading';

export default Loading;
