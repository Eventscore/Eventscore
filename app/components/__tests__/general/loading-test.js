/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '@components/general/Loading';

it('Loading renders correctly', () => {
  const tree = renderer.create(
    <Loading />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Loading w/ text renders correctly', () => {
  const tree = renderer.create(
    <Loading text={'Checking for Updates'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
