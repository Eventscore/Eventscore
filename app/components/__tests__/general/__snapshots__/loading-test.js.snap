// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      false,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#AAA"
    hidesWhenStopped={true}
    size="large"
  />
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    }
  />
</View>
`;

exports[`Loading w/ text renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      false,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#AAA"
    hidesWhenStopped={true}
    size="large"
  />
  <View
    style={
      Object {
        "height": 1,
        "left": 0,
        "marginTop": 9,
        "right": 0,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    onPress={null}
    p={false}
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 18,
        },
      ]
    }
  >
    Checking for Updates
  </Text>
</View>
`;
